docs/glossary.md

```diff
@@ -0,0 +1,15 @@
+| colormap          | An array of colors used to map pixel data to the actual color values.                                                                                                                                                                                                                                                                       |
+| dock widget        | A widget that can be attached to the main napari viewer at a specified position.                                                                                                                                                                                                                                                            |
+| hook               | Hook is the place within napari where we “promise” to call functions created by external developers and installed by the user.                                                                                                                                                                                                              |
+| hook specification | A function signature that defines the API that a plugin developer must adhere to when writing the function that we promise to call somewhere in the napari codebase.                                                                                                                                                                        |
+| keybinding         | The key or key combination that causes some functions to occur.                                                                                                                                                                                                                                                                             |
+| labels             | The labels layer is an image where pixel values are integers. Each unique integer represents a unique object, feature or classification.                                                                                                                                                                                                    |
+| layer              | Layers are the viewable objects that can be added to napari viewer. Currently napari supports the following layers: image, labels, points, shapes, surface, tracks, vectors.                                                                                                                                                                |
+| mousebinding       | The specific mouse click that causes some functions to occur.                                                                                                                                                                                                                                                                               |
```

Comment:
In other programming environments such as C++ Qt, Windows Forms, etc, the terminology is "mouse-events", such as `onMousePress`, `onMouseMove`, `onMouseRelease`. Can we connect napari terminology to the terminology of those common platforms?